software/base/src/main/java/org/apache/brooklyn/entity/brooklynnode/BrooklynNodeSshDriver.java

```diff
@@ -224,8 +231,18 @@ public void customize() {
             uploadFileContents(brooklynLocalPropertiesContents, brooklynLocalPropertiesUri, brooklynLocalPropertiesRemotePath);
         }
 
+        // Upload a local-catalog.bom if required
+        if (brooklynCatalogInitialBomContents != null || brooklynCatalogInitialBomUri != null) {
+            uploadFileContents(brooklynCatalogInitialBomContents, brooklynCatalogInitialBomUri, brooklynCatalogInitialBomRemotePath);
+        }
+
         // Override the ~/.brooklyn/catalog.xml if required
         if (brooklynCatalogContents != null || brooklynCatalogUri != null) {
+            log.warn("Deprecated use of config"
```

Comment:
Add `Deprecated since x.x.x`. Looking at the config keys it would be `0.7.0`, but since that's not user visible might make sense to set it to 0.9.0.